File: spk/API/Entities/Course.cs
namespace API.Entities;

public class Course : BaseEntity
{
    public Guid LicenseId { get; set; }

    public License? License { get; set; }

    public string Name { get; set; } = string.Empty;

    public string Slug { get; set; } = string.Empty;

    public string? Description { get; set; }

    public int Order { get; set; }

    public ICollection<Topic> Topics { get; set; } = new List<Topic>();

    public ICollection<SourceDocument> SourceDocuments { get; set; } = new List<SourceDocument>();
}

File: Client/src/models/course.ts
export interface Course {
  id: string
  licenseId: string
  name: string
  slug: string
  description?: string | null
  order: number
  topicCount: number
}

export interface CreateCourse {
  licenseId: string
  name: string
  slug: string
  description?: string | null
  order: number
}

export type UpdateCourse = CreateCourse
